### DISCUSSION TITLE
Undershoot in Generator torque

### DISCUSSION THREAD

**Discussion by Marcodep23** [NatLabRockies/ROSCO#266] [Q&A]
Hi everyone,
I'm simulating a scaled model (rotor diameter = 1.2m) in below rated region with the TSR tracking control technique with a step wind input (from 3m/s to 5.7m/s, that is the rated wind speed). 
The generator torque shows these undershoots at each step, indipendently from the gains of PI control. Any idea?
Thank you all!
[WT12_OPENFast.zip](https://github.com/NREL/ROSCO/files/12879240/WT12_OPENFast.zip)
<img width="527" alt="undershootGenTq" src="https://github.com/NREL/ROSCO/assets/147702381/3116fe49-e308-4fe5-be1b-b6ca789bebdd">

**Comment by dzalkind** [comment]
I have seen this if you use the exact wind speed, rather than the wind speed estimate (WSE) to calculate the generator speed set point when tracking the tip speed ratio.

I see from your DISCON that the WSE is disabled and the filter on the wind speed measurement is fast.  When this is the case, the step change in wind speed creates a step change in the set point and the torque drops quickly to speed up the rotor.

You can either enable the WSE (WE_Mode 2), decrease the filter frequency on the wind speed (F_WECornerFreq), or decrease the torque control bandwidth (omega_vs).
